blocks/library/code/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import TextareaAutosize from 'react-autosize-textarea';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlock, query } from '../../api';
+
+const { prop } = query;
+
+registerBlock( 'core/code', {
+	title: wp.i18n.__( 'Code' ),
+
+	icon: 'editor-code',
+
+	category: 'common',
+
+	attributes: {
+		content: prop( 'code', 'textContent' ),
+	},
+
+	edit( { attributes, setAttributes, setFocus } ) {
+		return (
+			<TextareaAutosize
+				defaultValue={ attributes.content }
```

Comment:
I think this could have been `value=` instead to treat the input as a [controlled component](https://facebook.github.io/react/docs/forms.html#controlled-components), ensuring that if, for whatever reason, the value were assigned explicitly, the textarea would be updated to reflect it.